plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -627,20 +627,44 @@ type ServiceAffinity struct {
 	labels        []string
 }
 
-func NewServiceAffinityPredicate(podLister algorithm.PodLister, serviceLister algorithm.ServiceLister, nodeInfo NodeInfo, labels []string) algorithm.FitPredicate {
+// serviceAffinityPrecomputation should be run once by the scheduler before looping through the Predicate.  It is a helper function that
+// only should be referenced by NewServiceAffinityPredicate.
+func (s *ServiceAffinity) serviceAffinityPrecomputation() PredicateMetadataModifier {
+	return func(pm *predicateMetadata) {
```

Comment:
I don't understand. You can simply have:
```
func (s *ServiceAffinity) serviceAffinityPrecomputation(pm *predicateMetadata) {
  // the body of this func
}
```

and this will work. It will be modifying the given pm.